ktor-client/ktor-client-core/common/src/io/ktor/client/features/websocket/builders.kt

```diff
@@ -39,14 +41,24 @@ suspend fun HttpClient.webSocketSession(
 suspend fun HttpClient.webSocket(
     request: HttpRequestBuilder.() -> Unit, block: suspend DefaultClientWebSocketSession.() -> Unit
 ) {
-    val session = webSocketSession(request)
-
-    try {
-        session.block()
-        session.close()
-    } catch (cause: Throwable) {
-        session.closeExceptionally(cause)
-        throw cause
+    val session = request<HttpStatement> {
+        url {
+            protocol = URLProtocol.WS
+            port = protocol.defaultPort
+        }
+        request()
+    }
+
+    session.receive<DefaultClientWebSocketSession, Unit> {
+        // Workaround: https://youtrack.jetbrains.com/issue/KT-34041
```

Comment:
This should already work well with eap2 so we probably need to eliminate this workaround